$MirOS: ports/devel/libgtop/patches/patch-sysdeps_freebsd_proctime_c,v 1.1.7.1 2005/03/18 15:44:51 tg Exp $
--- sysdeps/freebsd/proctime.c.orig	Tue Jan  2 23:50:05 2007
+++ sysdeps/freebsd/proctime.c	Tue Jun 26 22:46:45 2007
@@ -117,7 +117,7 @@ glibtop_get_proc_time_p (glibtop *server
 			 pid_t pid)
 {
 	struct kinfo_proc *pinfo;
-#if (defined(__NetBSD__) && (__NetBSD_Version__ >= 104000000)) || (defined(OpenBSD) && (OpenBSD >= 199912))
+#if (defined(__NetBSD__) && (__NetBSD_Version__ >= 104000000)) || (defined(OpenBSD) && (OpenBSD >= 199912)) || defined(__MirBSD__)
 	register struct rusage *rup;
 #else
 	struct user *u_addr = (struct user *)USRSTACK;
@@ -153,7 +153,7 @@ glibtop_get_proc_time_p (glibtop *server
 	buf->frequency = 1000000;
 	buf->flags = _glibtop_sysdeps_proc_time;
 
-#if (defined(__NetBSD__) && (__NetBSD_Version__ >= 104000000)) || (defined(OpenBSD) && (OpenBSD >= 199912))
+#if (defined(__NetBSD__) && (__NetBSD_Version__ >= 104000000)) || (defined(OpenBSD) && (OpenBSD >= 199912)) || defined(__MirBSD__)
 	glibtop_suid_enter (server);
 
 	if (kvm_read (server->machine.kd,
